src/omv/modules/py_display.c

```diff
@@ -105,6 +105,7 @@ STATIC mp_obj_t py_display_backlight(uint n_args, const mp_obj_t *args) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(py_display_backlight_obj, 1, 2, py_display_backlight);
 
 STATIC mp_obj_t py_display_write(uint n_args, const mp_obj_t *args, mp_map_t *kw_args) {
+    fb_alloc_mark();
```

Comment:
Wouldn't it just be easier to `fb_alloc_mark` in `py_image_cobj` ? It seems one mark precedes every call to `py_image_cobj`.

Reply:
That can't be done because py_image_cobj is called multiple times inside a function. The right solution would be to just add fb_alloc_mark() and fb_alloc_free_till_mark() to whatever calls functions in micropython. This will globally remove the need for these anywhere else and would clean all this up.

Would you like to do that?

Reply:
I think everything eventually calls `mp_call_function_n_kw` to do the actual call, hooks can be added there, but this is also used to call everything, including callbacks from Python/C.

Reply:
https://github.com/openmv/micropython/pull/115

However, on thinking more about this we should probably just keep manually wrapping things.

I tested the PR out and the system still works. But, small function calls in a loop will see a pretty steep slow down. Like toggling a GPIO pin in a loop.

Reply:
> That can't be done because py_image_cobj is called multiple times inside a function

The only 2 functions that call it multiple times are `int_py_tf_segment` and `py_image_to`. For `int_py_tf_segment` it calls it on a list of image objects, so should be safe. For `py_image_to`, and all of the conversion functions that call it `to_xxx/compressed`, it doesn't make any sense to pass a string, why would you:

```Python
img.to_bitmap(copy="path/to_imag") 
```

The `copy=img` feature and arg_other stuff can be dropped, but if you want to keep it, we should assert that `copy` is Not a string.

Reply:
Yeah, I agree with dropping all the copy=img stuff. Not needed anymore.

As for it being called multiple times. I mean like, any function that supports a mask argument technically accepts a second image. And then like draw_image(). There are a ton of methods that support mask images. So, I moved the code around whenever there's more than 1 image object in use by the method.

Reply:
> I mean like, any function that supports a mask argument technically accepts a second image. And then like draw_image().

A second image does Not cause `fb_alloc_mark` to be called, only a path. Can you point out a specific function where `py_image_cobj` is called twice in a row such that `fb_alloc_mark` is called twice ?

Reply:
Difference for example. You could pass an image which could be a path and then a mask: https://github.com/openmv/openmv/blob/master/src/omv/modules/py_image.c#L2383